File: Week 1/10.c
#include <stdio.h>
#include <stdlib.h>

// Function to reverse the array
int* reverseArray(int a_count, int* a, int* result_count) {
    *result_count = a_count;

    int* reversed = malloc(a_count * sizeof(int));

    for (int i = 0; i < a_count; i++) {
        reversed[i] = a[a_count - 1 - i];
    }

    return reversed;
}

int main() {
    int arr_count;

    // Input: size of array
    printf("Enter number of elements: ");
    scanf("%d", &arr_count);

    int* arr = malloc(arr_count * sizeof(int));

    // Input: array elements
    printf("Enter %d integers:\n", arr_count);
    for (int i = 0; i < arr_count; i++) {
        scanf("%d", &arr[i]);
    }

    int res_count;
    int* res = reverseArray(arr_count, arr, &res_count);

    // Output: reversed array
    printf("Reversed array:\n");
    for (int i = 0; i < res_count; i++) {
        printf("%d ", res[i]);
    }
    printf("\n");

    // Free allocated memory
    free(arr);
    free(res);

    return 0;
}

File: Week 1/10.py
#!/bin/python3  # Environment shebang for Unix-based systems

import math
import os
import random
import re
import sys

# Function to reverse the array
def reverseArray(a):
    return a[::-1]  # List slicing to reverse the list

if __name__ == '__main__':
    arr_count = int(input().strip())  # Input: number of elements
    
    arr = list(map(int, input().rstrip().split()))  # Input: array elements as list of ints

    res = reverseArray(arr)  # Function call to reverse

    print(' '.join(map(str, res)))  # Output: space-separated reversed list
